CSE.BL/BinaryFileManager.cs

```
@@ -1,31 +1,75 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.IO;
-using System.Text;
 
 namespace CSE.BL
 {
-    class BinaryFileManager
+    public class BinaryFileManager
     {
-        /* Writes the given object instance to a binary file.
-         * Parameter "append" - if value = false, the file will be overwritten if it already exists. If true the contents will be appended to the file.*/
-
-        public static void WriteToBinaryFile<T>(string filePath, T objectToWrite, bool append = false)
+        /* Writes the given object instance to a binary file.*/
+        public static void WriteToBinaryFile<T>(string filePath, T objectToWrite)
         {
-            using (Stream stream = File.Open(filePath, append ? FileMode.Append : FileMode.Create))
+            try
+            {
+                using (Stream stream = File.Open(filePath, FileMode.Create))
+                {
+                    var binaryFormatter = new System.Runtime.Serialization.Formatters.Binary.BinaryFormatter();
+                    binaryFormatter.Serialize(stream, objectToWrite);
+                }
+            } catch (DirectoryNotFoundException)
```

why are both exceptions swallowed?